lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -180,6 +180,12 @@ sub schedule_iso {
 # make sure that the DISTRI is lowercase
 sub _distri_key { lc(shift->{DISTRI}) }
 
+sub _delete_prefixed_args_storing_info_about_product_itself ($args) {
+    for my $arg (keys %$args) {
+        delete $args->{$arg} if index($arg, '__') == 0;
```

Comment:
`$arg =~ m/^__/` should be faster than `index()` because the `^` in the regex will only search at the beginning, and `index()` will search the whole string.
alternatively `substr($arg, 0, 2) eq '__'`